### PR TITLE
Add class grades feature

### PR DESCRIPTION
**Pull request by ngeeyonglim**
Added the attendance and the class participation parts of the class grades feature.

### CODE REVIEW

File: src/main/java/seedu/address/model/student/grades/AttendanceTracker.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Arrays;
+
+import seedu.address.commons.core.index.Index;
+
+/**
+ * Represents a Student's AttendanceTracker grades
+ * in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAttendanceTracker(String)}
```
Comment (YeoBohShin) [inline src/main/java/seedu/address/model/student/grades/AttendanceTracker.java:12]: Should be isValidAttendanceTracker(int)

File: src/main/java/seedu/address/model/student/grades/ClassParticipationTracker.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
+package seedu.address.model.student.grades;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import seedu.address.commons.core.index.Index;
+
+import java.util.Arrays;
```
Comment (YeoBohShin) [inline src/main/java/seedu/address/model/student/grades/ClassParticipationTracker.java]: Should exchange line 6 and 8 for checkstyle

File: src/main/java/seedu/address/model/student/grades/ClassParticipationTracker.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+/**
+ * Represents a Student's class participation grades
+ * in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidClassParticipationTracker(String)}
```
Comment (YeoBohShin) [inline src/main/java/seedu/address/model/student/grades/ClassParticipationTracker.java:13]: Should be isValidClassParticipationTracker(int)

File: src/main/java/seedu/address/model/student/grades/ClassParticipationTracker.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+     */
+    public ClassParticipationTracker(int numOfTut) {
+        requireNonNull(numOfTut);
+        checkArgument(isValidClassPart(numOfTut), MESSAGE_CONSTRAINTS);
+        classPartList = new ClassParticipation[numOfTut];
+        Arrays.fill(classPartList, new ClassParticipation());
+    }
+
+    /**
+     * Returns true if a given string is a valid number of tutorials.
```
Comment (YeoBohShin) [inline src/main/java/seedu/address/model/student/grades/ClassParticipationTracker.java:35]: Should be given Integer

File: src/main/java/seedu/address/model/student/grades/ClassParticipationTracker.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof ClassParticipationTracker)) {
+            return false;
+        }
+
+        ClassParticipationTracker otherClassParticipationTracker = (ClassParticipationTracker) other;
+        return Arrays.equals(classPartList,otherClassParticipationTracker.classPartList);
```
Comment (YeoBohShin) [inline src/main/java/seedu/address/model/student/grades/ClassParticipationTracker.java]: Needs whitespace after the comma

**Comment by Cikguseven** [comment]
LGTM

**Pull request merged**